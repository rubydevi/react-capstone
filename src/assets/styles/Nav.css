.menu {
  background-color: #ec4c8a;
  color: #fff;
  display: flex;
  padding: 3%;
  justify-content: space-between;
}

.back-button {
  background-color: transparent;
  outline: none;
  border: none;
  color: #fff;
  font-size: 1.3em;
  cursor: pointer;
}
