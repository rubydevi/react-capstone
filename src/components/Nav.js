import React from 'react';
import { useNavigate } from 'react-router-dom';
import { FaUser, FaChevronLeft } from 'react-icons/fa';
import '../assets/styles/Nav.css';

const Nav = () => {
  const navigate = useNavigate();

  const handleBackButtonClick = () => {
    navigate('/');
  };

  return (
    <ul className="menu">
      <li className="logo">
        <button type="button" className="back-button" onClick={handleBackButtonClick}>
          <FaChevronLeft />
        </button>

      </li>
      <li className="logo">
        <h3 className="logo-text">Pokemon</h3>
      </li>
      <li className="icons">
        <FaUser />
      </li>
    </ul>
  );
};

export default Nav;
