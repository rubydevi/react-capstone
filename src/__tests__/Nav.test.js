import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Nav from '../components/Nav';

test('Nav component snapshot', () => {
  const tree = renderer
    .create(
      <Router>
        <Nav />
      </Router>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
