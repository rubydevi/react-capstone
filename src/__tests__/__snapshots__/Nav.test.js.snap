// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component snapshot 1`] = `
<ul
  className="menu"
>
  <li
    className="logo"
  >
    <button
      className="back-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
        />
      </svg>
    </button>
  </li>
  <li
    className="logo"
  >
    <h3
      className="logo-text"
    >
      Pokemon
    </h3>
  </li>
  <li
    className="icons"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
      />
    </svg>
  </li>
</ul>
`;
